File: Queue/QueueLinkedList.py
import sys 
class Node:
    def __init__(self, value):
        self.value = value
        self.next = None

class Queue:
    def __init__(self):
        self.head = None
        self.tail = None
    
    def isEmpty(self):
        return self.head == None
    
    def peek(self):
        return self.head
    
    def contains(self, value):
        iterator = self.head
        while (iterator != self.tail and iterator.value != value):
            iterator = iterator.next
        return iterator.value == value
    
    def enqueue(self, X):
        if (self.tail==None): 
            self.head = X
            self.tail = X
            return     
        self.tail.next = X
        self.tail = X

    def dequeue(self):
        if (self.isEmpty()): 
            return
        temp = self.head
        self.head = self.head.next
        
        if (self.head == None):
            self.tail = None
        
        return temp
        
Q = Queue()

# <| Test Block |>
# Q.enqueue(Node(2))
# Q.enqueue(Node(3))
# Q.enqueue(Node(1))
# print(Q.peek().value)
# print(Q.contains(3))
# print(Q.contains(4))

# <| Problem Block |>     
N = int(input())

for i in range(N):
    str = input().split()
    if str[0][0]=='E':
        Q.enqueue(Node(int(str[1])))
    elif str[0][0]=="D":
        print(Q.dequeue().value)

File: Queue/QueueLinkedList.cpp
#include <bits/stdc++.h>
using namespace std;

struct Node {
    int data;
    Node * next;
};

typedef Node * Queue;

int dequeue(Queue &head, Queue &tail) {
    if (head == tail) {
        int val = head->data;
        delete head;
        return val;
    } else {
        Node * q = head;
        while (q->next != tail) {q=q->next;}
        int val = tail->data;
        delete tail;
        q->next = NULL;
        tail = q;
        return val;
    }
}

void enqueue(Queue &head, Queue &tail, int x) {
    Node * q = new Node;
    q-> next = head;
    q-> data = x;
    if (!head) tail = q;
    head = q;
}

int main() {
    int n=0;
    cin>>n;
    string b;
    getline(cin,b);
    Queue head = NULL, tail = NULL;
    for (int i=0;i<n;i++) {
        string str="";
        getline(cin,str);
        if (str[0]=='E') {
            enqueue(head, tail, stoi(str.substr(2,str.find('\n'))));
        } else if (str[0]=='D') {
            cout<<dequeue(head, tail)<<endl;
        }
    }
    return 0;
}
